modules/library/referencing/src/main/java/org/geotools/referencing/operation/transform/WarpTransform2D.java

```diff
@@ -411,14 +404,14 @@ public ParameterValueGroup getParameterValues() {
             final WarpPolynomial poly = (WarpPolynomial) warp;
             final ParameterValue[] p = new ParameterValue[7];
             int c = 0;
-            p[c++] = new Parameter(Provider.DEGREE,   Integer.valueOf(poly.getDegree()));
-            p[c++] = new Parameter(Provider.X_COEFFS, poly.getXCoeffs());
-            p[c++] = new Parameter(Provider.Y_COEFFS, poly.getYCoeffs());
+            p[c++] = new Parameter(WarpTransform2DProvider.DEGREE,   Integer.valueOf(poly.getDegree()));
+            p[c++] = new Parameter(WarpTransform2DProvider.X_COEFFS, poly.getXCoeffs());
+            p[c++] = new Parameter(WarpTransform2DProvider.Y_COEFFS, poly.getYCoeffs());
             float s;
-            if ((s=poly.getPreScaleX ()) != 1) p[c++] = new Parameter(Provider. PRE_SCALE_X, s);
-            if ((s=poly.getPreScaleY ()) != 1) p[c++] = new Parameter(Provider. PRE_SCALE_Y, s);
-            if ((s=poly.getPostScaleX()) != 1) p[c++] = new Parameter(Provider.POST_SCALE_X, s);
-            if ((s=poly.getPostScaleY()) != 1) p[c++] = new Parameter(Provider.POST_SCALE_Y, s);
+            if ((s=poly.getPreScaleX ()) != 1) p[c++] = new Parameter(WarpTransform2DProvider. PRE_SCALE_X, s);
+            if ((s=poly.getPreScaleY ()) != 1) p[c++] = new Parameter(WarpTransform2DProvider. PRE_SCALE_Y, s);
+            if ((s=poly.getPostScaleX()) != 1) p[c++] = new Parameter(WarpTransform2DProvider.POST_SCALE_X, s);
+            if ((s=poly.getPostScaleY()) != 1) p[c++] = new Parameter(WarpTransform2DProvider.POST_SCALE_Y, s);
```

Comment:
I do wonder if I should make a useless static Provider class so that these public parameters do not have to change?
{code}
/** Placeholder interface used for parameter definition */
static interface Provider {
        public static final ParameterDescriptor X_COEFFS = new DefaultParameterDescriptor("xCoeffs", float[].class, null, null);
        ...
}
{code}

The actual provider had to move to a separate class (and use reflection) to avoid crashing SPI factory lookup.